src/language/mongoDBService.ts

```diff
@@ -283,7 +283,7 @@ export default class MongoDBService {
           );
         }
 
-        worker.terminate().then(() => {
+        void worker.terminate().then(() => {
```

Comment:
Can't you use `async/await` also in the callback?

```
 worker.on('message', async (response: [Error, CompletionItem[] | []]) => { ... })
```

And then just `await` here .. i'm also not sure if this `terminate` is so fail safe, won't we have an unhandled promise rejection if that fails?

Reply:
I think WorkerThreads expect a void function here and do not allow Promise.

Reply:
That can be worked around, i don't think we should let that force us to use a particular style of promises.

Reply:
Personally I would also lean towards what @mcasimir said because if it throws there will be an unhandled rejection. Alternatively you can .catch() and do something with the error.

Thinking about it some more that try/catch won't really catch much since all the action happens in a callback. Personally I'd remove it and maybe audit usage of try/catch when there are promises and callbacks inside them.

Reply:
I just audited all the try/catch calls outside of tests, looking for .then() or callbacks that might be asynchronous and didn't spot any that stood out to me other than the four in this file.